<launch>

  <arg name="enable_debug"                 default="true"/>
  <arg name="sim_control"                  default="false"/>
  <arg name="enable_wf_control"            default="true"/>
  <arg name="enable_sf_control"            default="false"/>
  <arg name="enable_altitude_hold"         default="false"/>
  <arg name="enable_speed_regulation"      default="false"/>
  <arg name="enable_command_scaling"       default="false"/>
  <arg name="use_observed_shapes"          default="true"/>

  <arg name="enable_teleop"                default = "1"/>

  <!-- Launch the config file, which contains controller gains and other relevant parameters -->
  <rosparam file="$(find nearness_control)/cfg/NearnessControl3D.yaml" command="load" ns="nearness_control_3d_node"/>

  <!-- Launch the nearness control node -->
  <node pkg="nearness_control" type="nearness_control_3d_node" name="nearness_control_3d" output="screen">

      <!-- SUBSCRIBED TOPICS -->
      <remap from="nearness_control_3d/points"   to="ouster_points"/>
      <remap from="nearness_control_3d/odometry" to="/odometry/ground_truth"/>
      <!-- <remap from="nearness_controller_3d/control_commands" to="cmd_vel"/> -->
      <remap from="nearness_control_3d/joy" to="joy"/>

      <!-- Enable switches -->
      <param name="enable_debug"             type="bool"       value="$(arg enable_debug)"/>
      <param name="sim_control"              type="bool"       value="$(arg sim_control)"/>
      <param name="enable_wf_control"        type="bool"       value="$(arg enable_wf_control)"/>
      <param name="enable_sf_control"        type="bool"       value="$(arg enable_sf_control)"/>
      <param name="enable_altitude_hold"     type="bool"       value="$(arg enable_altitude_hold)"/>
      <param name="enable_speed_regulation"  type="bool"       value="$(arg enable_speed_regulation)"/>
      <param name="enable_command_scaling"   type="bool"       value="$(arg enable_command_scaling)"/>
      <param name="use_observed_shapes"      type="bool"       value="$(arg use_observed_shapes)"/>

  </node>

</launch>
